packages/discovery/src/discovery/output/saveDiscoveryResult.ts

```diff
@@ -63,6 +102,141 @@ export async function saveDiscoveryResult(
   }
 }
 
+async function saveFlatSources(
+  rootPath: string,
+  results: Analysis[],
+  logger: DiscoveryLogger,
+  options: {
+    rootFolder?: string
+    sourcesFolder?: string
+    discoveryFilename?: string
+  },
+): Promise<void> {
+  const flatSourcesFolder = `.flat${options.sourcesFolder ?? ''}`
+  const flatSourcesPath = posix.join(rootPath, flatSourcesFolder)
+  const allContractNames = results.map((c) =>
```

Comment:
I agree that saving sources grew too much. I wanted to refactor it but decided against it to cut development time. I'll plan and talk about separating it into three steps instead of two.

Linear issue ID: [L2B-4185](https://linear.app/l2beat/issue/L2B-4185/simplify-the-architecture-of-saving-source-files-in-discovery)